.github/workflows/ci-unit-tests.yml

```diff
@@ -22,9 +22,15 @@ jobs:
     - name: Run unit tests
       run: make test-ci
 
+    - name: golangci-lint
```

Comment:
I ended up placing the lint step within the unit-tests, which is similar behaviour to the Makefile. However, I feel it would be nicer as a separate job in order to enable concurrency, but am [getting errors relating to undefined types](https://github.com/jaegertracing/jaeger/runs/3454770737?check_suite_focus=true). Oddly, testing on my fork doesn't have this issue.

I believe it's the same config as is with #2602 and does not appear to have encountered this problem. Maybe @daxmc99 has some ideas about this?

Reply:
It might be worth noting that (_I think_) those "undefined" errors could be due to not having run `go build` on jaeger, which explains why moving this step as part of the unit tests job works.